src/language/concurrency.md

```diff
@@ -144,80 +171,107 @@ void main() [!async!] {
 {% endprettify %}
 
 The `main()` function uses the `await` keyword in front of `_readFileAsync()`
-to let other Dart code (such as event handlers) use the CPU
-while native code (file I/O) executes.
-Using `await` also has the effect of
-converting the `Future<String>` returned by `_readFileAsync()` into a `String`.
-As a result, the `contents` variable has the implicit type `String`.
+to let other Dart code (such as event handlers) use the CPU while native code
+(file I/O) executes. Using `await` also has the effect of converting
+the `Future<String>` returned by `_readFileAsync()` into a `String`. As a
+result, the `contents` variable has the implicit type `String`.
 
 {{site.alert.note}}
-  The `await` keyword works only in functions that
-  have `async` before the function body.
+The `await` keyword works only in functions that have `async` before the
+function body.
 {{site.alert.end}}
 
-As the following figure shows,
-the Dart code pauses while `readAsString()` executes non-Dart code,
-in either the Dart virtual machine (VM) or the operating system (OS).
+As the following figure shows, the Dart code pauses while `readAsString()`
+executes non-Dart code, in either the Dart runtime or the operating. 
 Once `readAsString()` returns a value, Dart code execution resumes.
 
-![Flowchart-like figure showing app code executing from start to exit, waiting for native I/O in between](/assets/img/language/concurrency/basics-await.png)
+![Flowchart-like figure showing app code executing from start to exit, waiting
+for native I/O in between](/assets/img/language/concurrency/basics-await.png)
 
-If you'd like to learn more about using `async`, `await`, and futures,
-visit the [asynchronous programming codelab][].
+### Streams
 
-[asynchronous programming codelab]: /codelabs/async-await
+Dart also supports asynchronous code in the form of streams. Streams
+provide values in the future and repeatedly over time. A promise to provide a
+series of `int` values over time has the type `Stream<int>`.
 
+In the following example, the stream created with `Stream.periodic`
+repeatedly emits a new `int` value every second.
 
-## How isolates work
+<?code-excerpt "lib/stream_syntax.dart"?>
+```dart
+Stream<int> stream = Stream.periodic(const Duration(seconds: 1), (i) => i * i);
+```
+
+#### await-for and yield
```

Comment:
This is a great section! I didn't know anything about streams and found them intimidating before and now they seem simple and approachable